### PR TITLE
feat: Add reward points to Customers

### LINKED ISSUE #42: As a small business owner, I can set the amount of reward points a customer has
so that I can track my customers' points

### PR DESCRIPTION
**Pull request by Infrix**
## Changes
* [Model] Add Points attribute for Customer
* [Logic] Add setpoints command to assign `Points` to a Customer
* [Storage] Include storage logic for `Points`
* Add tests for model, logic and storage of `Points`

Closes #42

### CODE REVIEW

File: src/main/java/seedu/loyaltylift/storage/JsonAdaptedCustomer.java [hunk 1]
```diff
@@ -119,7 +125,16 @@ public Customer toModelType() throws IllegalValueException {
...
 
         final Set<Tag> modelTags = new HashSet<>(customerTags);
-        return new Customer(modelCustomerType, modelName, modelPhone, modelEmail, modelAddress, modelTags);
+
+        if (points == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Points.class.getSimpleName()));
+        }
+        if (!Address.isValidAddress(address)) {
+            throw new IllegalValueException(Address.MESSAGE_CONSTRAINTS);
+        }
```
Comment (CloudHill) [inline src/main/java/seedu/loyaltylift/storage/JsonAdaptedCustomer.java:134]: Should be checking for `Points` here right?

File: src/main/java/seedu/loyaltylift/model/customer/Points.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        return other == this // short circuit if same object
+                || (other instanceof Points // instanceof handles nulls
+                && value.equals(((Points) other).value)); // state check
+    }
+
+    @Override
+    public int hashCode() {
+        return value.hashCode();
+    }
+}
```
Comment (CloudHill) [inline src/main/java/seedu/loyaltylift/model/customer/Points.java:54]: Maybe add `isValidPoints` to ensure that points are not negative? Can refer to other attributes

**Review by CloudHill — CHANGES_REQUESTED** [review]
Just a couple small changes in `Points` and `JsonAdaptedCustomer`

There are also a bunch of lines not covered by tests, but I guess we don't have to deal with them now

File: src/main/java/seedu/loyaltylift/storage/JsonAdaptedCustomer.java (hunk 1 above)
Comment (Infrix) [reply src/main/java/seedu/loyaltylift/storage/JsonAdaptedCustomer.java:134]: yes, thanks for catching that bug, my bad

File: src/main/java/seedu/loyaltylift/model/customer/Points.java (hunk 2 above)
Comment (Infrix) [reply src/main/java/seedu/loyaltylift/model/customer/Points.java:54]: Done, points can only range from 0 to 999999

**Review by CloudHill — APPROVED** [review]

**Pull request merged**